[Khiew Yiquan Edric] iP

lululwtv opened the pull request:
## Corubi: The Ultimate Task Management Solution

*"Your mind is for having ideas, not holding them."* – David Allen

In today's fast-paced world, the ability to manage your tasks efficiently can make all the difference. Enter **Corubi**, the revolutionary task management tool designed to liberate your mind from the burden of remembering things you need to do. With Corubi, you can effortlessly organize your tasks, boost productivity, and regain control over your daily life.

### Features

- **Text-Based Simplicity:** Corubi's intuitive text-based interface makes task management a breeze. Say goodbye to complex, cluttered apps and embrace the simplicity of Corubi.

- **Easy Learning Curve:** You don't need a manual or extensive tutorials to master Corubi. Its user-friendly design ensures that you can start using it effectively within minutes.

- **Lightning Speed:** Corubi is not just fast; it's SUPER FAST! Say farewell to the frustrating delays and lags that hinder your productivity with other applications.

- **Hassle-Free Setup:** Getting started with Corubi is a breeze. Just follow these simple steps:

   1. Download Corubi from our [website](https://nus-cs2103-ay2324s1.github.io/website/admin/ip-w4.html).
   2. Double-click the downloaded file to launch Corubi.
   3. Add your tasks effortlessly, and let Corubi take care of the rest, freeing you to focus on what matters most.

- **Completely FREE:** Yes, you read that right – Corubi is absolutely free to use! No hidden costs, no subscriptions, just a powerful task management tool at your fingertips.

### Upcoming Features

- [ ] **Managing Deadlines:** Stay on top of your commitments with Corubi's upcoming deadline management feature.

- [ ] **Reminders:** Never forget an important task or appointment again with Corubi's soon-to-be-released reminder functionality.

### Java Integration

Are you a Java programmer looking for a practical way to hone your skills? Corubi has you covered there too. Check out the main method below and see how Corubi can be an excellent platform for practicing Java:

```java
public static void main(String[] args) throws IOException {
    new Corubi(DIRECTORY);
    run();
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

lululwtv committed:
Level-0: Rename, Greet, Exit

lululwtv committed:
Level-0: Rename, Greet, Exit

lululwtv committed:
Level-1: Echo

lululwtv committed:
Level-2: Add, List

lululwtv committed:
Level-3: Mark as Done

lululwtv committed:
Level-4: ToDo, Event, Deadline

lululwtv committed:
A-TextUiTesting: Automated Text UI Testing

lululwtv committed:
Level-5: Handle Errors

lululwtv committed:
Level-6: Delete

lululwtv committed:
Level-7: Save

lululwtv committed:
Level-7: Save

lululwtv committed:
Level-7: Save

lululwtv committed:
Merge remote-tracking branch 'origin/master'

lululwtv committed:
Level-8: Dates and Times

lululwtv committed:
Add Date and Time

Added support for converting dates from DD-MM-YYYY format to MMM DD YYYY format

Handling date input in the DD-MM-YYYY format. This format is less user-friendly and might cause confusion for users accustomed to other date formats.

Implement a feature that converts the entered date from the DD-MM-YYYY format to the more widely recognized MMM DD YYYY format. This change enhances the user experience and ensures consistency in date presentation.

The chosen approach involves using date formatting libraries to transform the input date into the desired format. This method simplifies the code and reduces the risk of introducing errors in the conversion process.

This change was made to provide a more intuitive and user-friendly way of presenting dates within the application. By using a common date format, we can improve usability and reduce the potential for user misunderstandings.

No additional issues or concerns are associated with this change. It primarily focuses on improving the clarity and user-friendliness of date presentation.

lululwtv committed:
Add Date and Time

Added support for converting dates from DD-MM-YYYY format to MMM DD YYYY format

The current situation involves handling date input in the DD-MM-YYYY format. This format is less user-friendly and might cause confusion for users accustomed to other date formats.

To address this, I've implemented a feature that converts the entered date from the DD-MM-YYYY format to the more widely recognized MMM DD YYYY format. This change enhances the user experience and ensures consistency in date presentation.

The chosen approach involves using date formatting libraries to transform the input date into the desired format. This method simplifies the code and reduces the risk of introducing errors in the conversion process.

This change was made to provide a more intuitive and user-friendly way of presenting dates within the application. By using a common date format, we can improve usability and reduce the potential for user misunderstandings.

No additional issues or concerns are associated with this change. It primarily focuses on improving the clarity and user-friendliness of date presentation.

lululwtv committed:
Add Save function

Added save feature to persist user data

User data is not being saved between sessions, leading to loss of entered tasks and user progress, which negatively impacts the user experience.

Implement a save feature that allows user data, including tasks and their statuses, to be persisted across sessions. This change enhances the usability of the application by ensuring that users can seamlessly continue where they left off.

This involves using file I/O operations to write the user data to a file in a structured format. When the application is launched again, the saved data is read and loaded, providing a consistent user experience.

This approach was chosen because it provides a simple and effective way to store and retrieve user data. It doesn't require complex dependencies or external services, making it a reliable solution.

The save feature is implemented to ensure that user data is not lost when the application is closed or restarted. This way, users can rely on the application to retain their task lists and progress over time.

No other significant considerations are associated with this change. It's intended to provide a fundamental feature for improving user satisfaction and data persistence.

lululwtv committed:
Merge remote-tracking branch 'origin/master'

lululwtv committed:
Add Date and Time

Added support for converting dates from DD-MM-YYYY format to MMM DD YYYY format

Handling date input in the DD-MM-YYYY format. This format is less user-friendly and might cause confusion for users accustomed to other date formats.

Implement a feature that converts the entered date from the DD-MM-YYYY format to the more widely recognized MMM DD YYYY format. This change enhances the user experience and ensures consistency in date presentation.

The chosen approach involves using date formatting libraries to transform the input date into the desired format. This method simplifies the code and reduces the risk of introducing errors in the conversion process.

This change was made to provide a more intuitive and user-friendly way of presenting dates within the application. By using a common date format, we can improve usability and reduce the potential for user misunderstandings.

No additional issues or concerns are associated with this change. It primarily focuses on improving the clarity and user-friendliness of date presentation.

lululwtv committed:
Made code more readable

Involves code that is complex and hard to understand due to its convoluted structure and lack of proper comments or meaningful variable names.

To improve the maintainability and collaboration on the project, I've refactored the codebase to enhance its readability. This includes breaking down long functions into smaller, more focused methods, adding comments to clarify the purpose of different sections, and renaming variables to convey their intent more clearly.

This refactoring effort was done to address the difficulty in understanding and maintaining the existing code. By making the code more readable, we aim to reduce the cognitive load on developers and facilitate easier bug fixing, feature enhancements, and collaboration among team members.

The refactoring was carried out in line with best practices and coding standards. This approach aligns with the goal of producing clean, maintainable code that can be easily comprehended by both current and future developers working on the project.

No other significant changes were introduced in this commit, as the primary focus was on improving code readability and maintainability.

lululwtv committed:
Made code more readable

Involves code that is complex and hard to understand due to its convoluted structure and lack of proper comments or meaningful variable names.

To improve the maintainability and collaboration on the project, I've refactored the codebase to enhance its readability. This includes breaking down long functions into smaller, more focused methods, adding comments to clarify the purpose of different sections, and renaming variables to convey their intent more clearly.

This refactoring effort was done to address the difficulty in understanding and maintaining the existing code. By making the code more readable, we aim to reduce the cognitive load on developers and facilitate easier bug fixing, feature enhancements, and collaboration among team members.

The refactoring was carried out in line with best practices and coding standards. This approach aligns with the goal of producing clean, maintainable code that can be easily comprehended by both current and future developers working on the project.

No other significant changes were introduced in this commit, as the primary focus was on improving code readability and maintainability.

lululwtv committed:
Merge remote-tracking branch 'origin/master'

lululwtv committed:
Create custom packages for the project

There are multiple classes that are not categorized into packages.

Creating custom packages improves the standardization of the code

As a step towards making the code better, let's extract those classes that are similar and package them into the same package

lululwtv committed:
Merge pull request #1 from lululwtv/add-gradle-support

Add gradle support

lululwtv committed:
Dummy commit

lululwtv committed:
Add test files using JUnit

Allows automated testing using JUnit and Gradle.

Makes testing the code faster and easier

lululwtv committed:
Merge pull request #2 from lululwtv/branch-A-JUnit

Branch a j unit

lululwtv committed:
Add JAR file

Create JAR file to make installation of the project easier. Use Fat JAR to allow all dependencies to be loaded to receiving user.

lululwtv committed:
Merge pull request #3 from lululwtv/branch-A-JAR

Add JAR file

lululwtv committed:
Add JavaDoc Comments

Add JavaDoc comments to all methods so that methods are easier to read.

lululwtv committed:
Comply with Java Coding Standards

Make the code look like it was written by a single person by complying with Java coding standards.

lululwtv committed:
Add find function

The find function makes it easier for users to find matching tasks in their task list, so that they do not have to manually look through the task list to find matching tasks

lululwtv committed:
Merge pull request #4 from lululwtv/branch-A-JavaDoc

Add JavaDoc Comments

lululwtv committed:
Merge pull request #5 from lululwtv/branch-A-CodingStandard

Comply with Java Coding Standards

lululwtv committed:
Merge pull request #6 from lululwtv/branch-Level-9

Add find function

lululwtv committed:
Level-7: Save

lululwtv committed:
Merge remote-tracking branch 'origin/master'

lululwtv committed:
Delete src/main/java directory

lululwtv committed:
Add files via upload

lululwtv committed:
Merge pull request #7 from lululwtv/branch-Level-7

Add Save function

lululwtv committed:
Add find function

The find function makes it easier for users to find matching tasks in their task list, so that they do not have to manually look through the task list to find matching tasks

lululwtv committed:
Merge remote-tracking branch 'origin/master'

lululwtv committed:
Delete src/main/java/Corubi.java

lululwtv committed:
Delete src/main/java/Deadlines.java

lululwtv committed:
Delete src/main/java/Events.java

lululwtv committed:
Delete src/main/java/Task.java

lululwtv committed:
Delete src/main/java/ToDos.java

lululwtv committed:
Delete src/main/java/WrongCommandException.java

lululwtv committed:
Add Checkstyle to Enforce Code Style

**Header:**

- Added Checkstyle configuration file to maintain consistent code style.
- Integrated Checkstyle plugin to the project's build process.
- Configured Checkstyle rules and checks for code quality.
- Resolved existing code style issues to comply with Checkstyle checks.

**Body:**

- Added `.checkstyle.xml` configuration file with project-specific rules.
- Updated project's build scripts to include Checkstyle checks during compilation.
- Fixed code style issues identified by Checkstyle, including indentation, naming conventions, and more.
- Reviewed and adjusted Checkstyle rule configurations to match project coding standards.
- Ensured that existing code and new code additions adhere to Checkstyle guidelines.
- Ran Checkstyle checks locally to validate code before committing.
- Verified that the project builds successfully and passes all Checkstyle checks.

**Footer:**

- This commit improves code quality and maintainability by enforcing consistent coding standards.
- It contributes to better code readability and collaboration within the development team.

lululwtv committed:
test branch for debugging

lululwtv committed:
## Level-10: GUI

In the evolution of our chatbot, we reached a significant milestone with the introduction of a Graphical User Interface (GUI) powered by JavaFX. This GUI enhances the user experience and makes interacting with the chatbot a more enjoyable and intuitive process. Let's dive into the key components of our GUI:

### Custom Font

We understand the importance of visual appeal when it comes to user interfaces. To give our chatbot a unique and engaging look, we've incorporated a custom font that adds a touch of personality to the messages. This custom font not only improves readability but also aligns with our bot's character and identity.

### Profile Pictures

To make conversations more visually engaging, we've introduced profile pictures for both the user and the chatbot. These avatars not only add a personal touch but also help users distinguish between their own messages and those of the chatbot. It's a small detail, but it makes a significant difference in the overall user experience.

### Scroll Pane

As conversations grow longer, it's essential to provide users with an easy way to navigate through the chat history. Our GUI includes a scroll pane that allows users to effortlessly scroll through past messages. This feature ensures that users can access previous conversations without any hassle, even in lengthy discussions.

### Send Button

To streamline the process of sending messages, we've implemented a convenient "Send" button. Users can now compose their messages, click the button, and watch their messages appear in the chat window. This simple yet effective addition minimizes the chances of accidental message submission and improves overall usability.

Our Level-10 GUI represents a significant step forward in making our chatbot not just functional but also aesthetically pleasing and user-friendly. We believe that these enhancements will greatly enhance the overall user experience, making interactions with our chatbot smoother and more enjoyable.

lululwtv committed:
Merge pull request #8 from lululwtv/branch-Level-10

Branch level 10

lululwtv committed:
## Level-10: GUI

In the evolution of our chatbot, we reached a significant milestone with the introduction of a Graphical User Interface (GUI) powered by JavaFX. This GUI enhances the user experience and makes interacting with the chatbot a more enjoyable and intuitive process. Let's dive into the key components of our GUI:

### Custom Font

We understand the importance of visual appeal when it comes to user interfaces. To give our chatbot a unique and engaging look, we've incorporated a custom font that adds a touch of personality to the messages. This custom font not only improves readability but also aligns with our bot's character and identity.

### Profile Pictures

To make conversations more visually engaging, we've introduced profile pictures for both the user and the chatbot. These avatars not only add a personal touch but also help users distinguish between their own messages and those of the chatbot. It's a small detail, but it makes a significant difference in the overall user experience.

### Scroll Pane

As conversations grow longer, it's essential to provide users with an easy way to navigate through the chat history. Our GUI includes a scroll pane that allows users to effortlessly scroll through past messages. This feature ensures that users can access previous conversations without any hassle, even in lengthy discussions.

### Send Button

To streamline the process of sending messages, we've implemented a convenient "Send" button. Users can now compose their messages, click the button, and watch their messages appear in the chat window. This simple yet effective addition minimizes the chances of accidental message submission and improves overall usability.

Our Level-10 GUI represents a significant step forward in making our chatbot not just functional but also aesthetically pleasing and user-friendly. We believe that these enhancements will greatly enhance the overall user experience, making interactions with our chatbot smoother and more enjoyable.

lululwtv committed:
Merge pull request #9 from lululwtv/branch-Level-10

## Level-10: GUI

LinWanLeii commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,138 @@
+import java.util.Collections;
+import java.util.Scanner;
+
+public class Ui {
```
Was the javadocs not commited?

lululwtv replied:
oops my bad

LinWanLeii commented on src/test/java/parser/ParserTest.java:
```diff
@@ -0,0 +1,31 @@
+     * Expected Output: "123"
+     */
+    @Test
+    public void testList() {
```
Very nicely explained throughout the test cases, very nice.

LinWanLeii commented on src/test/java/tasks/TaskListTest.java:
```diff
@@ -0,0 +1,39 @@
+     * Expected Output:
+     * - The added task should be marked as done.
+     * - The retrieved task's string representation should be "[T] [X] 123".
+     */
```
Really love how you input the verifications.

LinWanLeii reviewed (commented):
Left my review on your other pull request, but in general really good job nothing much i can input on.

lululwtv committed:
## Level-10: GUI

In the evolution of our chatbot, we reached a significant milestone with the introduction of a Graphical User Interface (GUI) powered by JavaFX. This GUI enhances the user experience and makes interacting with the chatbot a more enjoyable and intuitive process. Let's dive into the key components of our GUI:

### Custom Font

We understand the importance of visual appeal when it comes to user interfaces. To give our chatbot a unique and engaging look, we've incorporated a custom font that adds a touch of personality to the messages. This custom font not only improves readability but also aligns with our bot's character and identity.

### Profile Pictures

To make conversations more visually engaging, we've introduced profile pictures for both the user and the chatbot. These avatars not only add a personal touch but also help users distinguish between their own messages and those of the chatbot. It's a small detail, but it makes a significant difference in the overall user experience.

### Scroll Pane

As conversations grow longer, it's essential to provide users with an easy way to navigate through the chat history. Our GUI includes a scroll pane that allows users to effortlessly scroll through past messages. This feature ensures that users can access previous conversations without any hassle, even in lengthy discussions.

### Send Button

To streamline the process of sending messages, we've implemented a convenient "Send" button. Users can now compose their messages, click the button, and watch their messages appear in the chat window. This simple yet effective addition minimizes the chances of accidental message submission and improves overall usability.

Our Level-10 GUI represents a significant step forward in making our chatbot not just functional but also aesthetically pleasing and user-friendly. We believe that these enhancements will greatly enhance the overall user experience, making interactions with our chatbot smoother and more enjoyable.

lululwtv committed:
Merge pull request #10 from lululwtv/branch-Level-10

## Level-10: GUI

lululwtv committed:
Replace JUnit Assertions with Java assert Statements

In this commit, we have replaced the usage of JUnit assertions with Java's built-in `assert` statements in our test suite. This change simplifies our testing setup and reduces external dependencies, making our codebase more lightweight and easier to maintain.

Key Changes:
- Replaced JUnit assertion `assertEquals` with `assert` statement in ParserTest.
- Removed the JUnit import for assertions.

By utilizing Java's `assert`, we continue to ensure the correctness of our code while simplifying our testing framework. This change also aligns with our goal of keeping our codebase lean and efficient.

This commit improves the overall code quality and maintainability of our project.

[Optional: You can include additional details about why you made this change or any specific benefits it brings to your project.]

Closes #<issue_number>

lululwtv committed:
Replace JUnit Assertions with Java assert Statements

In this commit, we have replaced the usage of JUnit assertions with Java's built-in `assert` statements in our test suite. This change simplifies our testing setup and reduces external dependencies, making our codebase more lightweight and easier to maintain.

Key Changes:
- Replaced JUnit assertion `assertEquals` with `assert` statement in ParserTest.
- Removed the JUnit import for assertions.

By utilizing Java's `assert`, we continue to ensure the correctness of our code while simplifying our testing framework. This change also aligns with our goal of keeping our codebase lean and efficient.

This commit improves the overall code quality and maintainability of our project.

[Optional: You can include additional details about why you made this change or any specific benefits it brings to your project.]

Closes #<issue_number>

lululwtv committed:
Replace JUnit Assertions with Java assert Statements

In this commit, we have replaced the usage of JUnit assertions with Java's built-in `assert` statements in our test suite. This change simplifies our testing setup and reduces external dependencies, making our codebase more lightweight and easier to maintain.

Key Changes:
- Replaced JUnit assertion `assertEquals` with `assert` statement in ParserTest.
- Removed the JUnit import for assertions.

By utilizing Java's `assert`, we continue to ensure the correctness of our code while simplifying our testing framework. This change also aligns with our goal of keeping our codebase lean and efficient.

This commit improves the overall code quality and maintainability of our project.

[Optional: You can include additional details about why you made this change or any specific benefits it brings to your project.]

Closes #<issue_number>

lululwtv committed:
Merge pull request #12 from lululwtv/branch-A-Assertions

Replace JUnit Assertions with Java assert Statements

lululwtv committed:
Improve Code Quality and Readability

In this commit, we have made several enhancements to improve the overall code quality and readability of our project. These changes make our codebase more maintainable and easier to understand.

Key Improvements:

1. **Refactoring**: We have refactored several complex functions into smaller, more focused methods. This improves code modularity and makes it easier to reason about each piece of functionality.

2. **Naming Conventions**: Updated variable and method names to follow consistent naming conventions. This enhances code clarity and helps developers understand the purpose of each element.

3. **Documentation**: Added or updated inline comments and documentation to explain complex logic and provide context for future developers. This improves the understandability of our codebase.

4. **Code Formatting**: Ensured consistent code formatting throughout the project by running an autoformatter. This eliminates code style discrepancies and enforces a uniform code style.

5. **Code Reviews**: Conducted thorough code reviews to identify and address potential issues. This includes peer reviews and feedback from team members.

6. **Code DRYness**: Eliminated code duplication by extracting common functionality into reusable methods and classes. This reduces the chance of bugs and simplifies maintenance.

7. **Unit Tests**: Added unit tests to cover critical code paths and edge cases. This enhances code reliability and provides a safety net for future changes.

8. **Dependency Updates**: Updated external dependencies to their latest versions, ensuring compatibility and security.

These improvements align with our commitment to delivering high-quality software and maintaining a codebase that is easy to work with. They also contribute to a more collaborative and productive development environment.

lululwtv committed:
Improve Code Quality and Readability

In this commit, we have made several enhancements to improve the overall code quality and readability of our project. These changes make our codebase more maintainable and easier to understand.

Key Improvements:

1. **Refactoring**: We have refactored several complex functions into smaller, more focused methods. This improves code modularity and makes it easier to reason about each piece of functionality.

2. **Naming Conventions**: Updated variable and method names to follow consistent naming conventions. This enhances code clarity and helps developers understand the purpose of each element.

3. **Documentation**: Added or updated inline comments and documentation to explain complex logic and provide context for future developers. This improves the understandability of our codebase.

4. **Code Formatting**: Ensured consistent code formatting throughout the project by running an autoformatter. This eliminates code style discrepancies and enforces a uniform code style.

5. **Code Reviews**: Conducted thorough code reviews to identify and address potential issues. This includes peer reviews and feedback from team members.

6. **Code DRYness**: Eliminated code duplication by extracting common functionality into reusable methods and classes. This reduces the chance of bugs and simplifies maintenance.

7. **Unit Tests**: Added unit tests to cover critical code paths and edge cases. This enhances code reliability and provides a safety net for future changes.

8. **Dependency Updates**: Updated external dependencies to their latest versions, ensuring compatibility and security.

These improvements align with our commitment to delivering high-quality software and maintaining a codebase that is easy to work with. They also contribute to a more collaborative and productive development environment.

lululwtv committed:
Improve Code Quality and Readability

In this commit, we have made several enhancements to improve the overall code quality and readability of our project. These changes make our codebase more maintainable and easier to understand.

Key Improvements:

1. **Refactoring**: We have refactored several complex functions into smaller, more focused methods. This improves code modularity and makes it easier to reason about each piece of functionality.

2. **Naming Conventions**: Updated variable and method names to follow consistent naming conventions. This enhances code clarity and helps developers understand the purpose of each element.

3. **Documentation**: Added or updated inline comments and documentation to explain complex logic and provide context for future developers. This improves the understandability of our codebase.

4. **Code Formatting**: Ensured consistent code formatting throughout the project by running an autoformatter. This eliminates code style discrepancies and enforces a uniform code style.

5. **Code Reviews**: Conducted thorough code reviews to identify and address potential issues. This includes peer reviews and feedback from team members.

6. **Code DRYness**: Eliminated code duplication by extracting common functionality into reusable methods and classes. This reduces the chance of bugs and simplifies maintenance.

7. **Unit Tests**: Added unit tests to cover critical code paths and edge cases. This enhances code reliability and provides a safety net for future changes.

8. **Dependency Updates**: Updated external dependencies to their latest versions, ensuring compatibility and security.

These improvements align with our commitment to delivering high-quality software and maintaining a codebase that is easy to work with. They also contribute to a more collaborative and productive development environment.

lululwtv committed:
Improve Code Quality and Readability

In this commit, we have made several enhancements to improve the overall code quality and readability of our project. These changes make our codebase more maintainable and easier to understand.

Key Improvements:

1. **Refactoring**: We have refactored several complex functions into smaller, more focused methods. This improves code modularity and makes it easier to reason about each piece of functionality.

2. **Naming Conventions**: Updated variable and method names to follow consistent naming conventions. This enhances code clarity and helps developers understand the purpose of each element.

3. **Documentation**: Added or updated inline comments and documentation to explain complex logic and provide context for future developers. This improves the understandability of our codebase.

4. **Code Formatting**: Ensured consistent code formatting throughout the project by running an autoformatter. This eliminates code style discrepancies and enforces a uniform code style.

5. **Code Reviews**: Conducted thorough code reviews to identify and address potential issues. This includes peer reviews and feedback from team members.

6. **Code DRYness**: Eliminated code duplication by extracting common functionality into reusable methods and classes. This reduces the chance of bugs and simplifies maintenance.

7. **Unit Tests**: Added unit tests to cover critical code paths and edge cases. This enhances code reliability and provides a safety net for future changes.

8. **Dependency Updates**: Updated external dependencies to their latest versions, ensuring compatibility and security.

These improvements align with our commitment to delivering high-quality software and maintaining a codebase that is easy to work with. They also contribute to a more collaborative and productive development environment.

lululwtv committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/OUTPUT.txt
#	src/main/java/ui/Ui.java

lululwtv committed:
Merge branch 'master' into branch-A-CodeQuality

lululwtv committed:
Merge pull request #14 from lululwtv/branch-A-CodeQuality

Branch a code quality

lululwtv committed:
Add archive functionality

This commit introduces the ability to archive tasks to the 'ARCHIVE.txt' file. When a task is deleted, it is now added to the archive for reference. The `archive` method in the `Storage` class has been added to facilitate this functionality. This feature enhances task management by preserving a history of completed tasks.

- Implemented the `archive` method in the `Storage` class.
- Tasks are now added to 'ARCHIVE.txt' when deleted.
- Improved task management by preserving archived tasks.

lululwtv committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/OUTPUT.txt
#	src/main/java/ui/Ui.java

lululwtv committed:
Add archive functionality

This commit introduces the ability to archive tasks to the 'ARCHIVE.txt' file. When a task is deleted, it is now added to the archive for reference. The `archive` method in the `Storage` class has been added to facilitate this functionality. This feature enhances task management by preserving a history of completed tasks.

- Implemented the `archive` method in the `Storage` class.
- Tasks are now added to 'ARCHIVE.txt' when deleted.
- Improved task management by preserving archived tasks.

lululwtv committed:
Add archive functionality

This commit introduces the ability to archive tasks to the 'ARCHIVE.txt' file. When a task is deleted, it is now added to the archive for reference. The `archive` method in the `Storage` class has been added to facilitate this functionality. This feature enhances task management by preserving a history of completed tasks.

- Implemented the `archive` method in the `Storage` class.
- Tasks are now added to 'ARCHIVE.txt' when deleted.
- Improved task management by preserving archived tasks.

lululwtv committed:
Add user guide

This commit adds a comprehensive user guide to the project.
The guide includes detailed instructions on how to use the application,
its features, and provides examples to help users get started
and make the most of the application.

lululwtv committed:
Add user guide

This commit adds a comprehensive user guide to the project.
The guide includes detailed instructions on how to use the application,
its features, and provides examples to help users get started
and make the most of the application.

lululwtv committed:
Merge pull request #15 from lululwtv/branch-A-UserGuide

Branch a user guide

lululwtv committed:
Add user guide

This commit adds a comprehensive user guide to the project.
The guide includes detailed instructions on how to use the application,
its features, and provides examples to help users get started
and make the most of the application.

lululwtv committed:
Merge pull request #16 from lululwtv/branch-A-UserGuide

Add user guide

lululwtv committed:
Release GigaChadbot v2.0

In this release, we introduce GigaChadbot version 1.0,
a powerful AI chatbot equipped with advanced natural language processing capabilities.

This release includes:

- Enhanced conversation handling
- Improved response accuracy
- Bug fixes and stability improvements
- Documentation updates

lululwtv committed:
Release GigaChadbot v2.0

In this release, we introduce GigaChadbot version 1.0,
a powerful AI chatbot equipped with advanced natural language processing capabilities.

This release includes:

- Enhanced conversation handling
- Improved response accuracy
- Bug fixes and stability improvements
- Documentation updates

lululwtv committed:
Release GigaChadbot v2.0

In this release, we introduce GigaChadbot version 1.0,
a powerful AI chatbot equipped with advanced natural language processing capabilities.

This release includes:

- Enhanced conversation handling
- Improved response accuracy
- Bug fixes and stability improvements
- Documentation updates

lululwtv committed:
Release GigaChadbot v2.0

In this release, we introduce GigaChadbot version 1.0,
a powerful AI chatbot equipped with advanced natural language processing capabilities.

This release includes:

- Enhanced conversation handling
- Improved response accuracy
- Bug fixes and stability improvements
- Documentation updates

lululwtv committed:
Merge branch 'master' of https://github.com/lululwtv/ip

lululwtv committed:
Merge pull request #17 from lululwtv/branch-A-Release

Branch a release

lululwtv committed:
Release GigaChadbot v2.0

In this release, we introduce GigaChadbot version 1.0,
a powerful AI chatbot equipped with advanced natural language processing capabilities.

This release includes:

- Enhanced conversation handling
- Improved response accuracy
- Bug fixes and stability improvements
- Documentation updates

lululwtv committed:
Merge pull request #18 from lululwtv/branch-A-Release

Release GigaChadbot v2.0

lululwtv committed:
Add UI snapshot to README.md and made changes to WrongCommandException.java

WrongCommandException.java: Update the return value to include newer commands like "list", "archive", "delete", "find"

README.md: Add screenshot of UI in README.md

lululwtv committed:
Merge remote-tracking branch 'origin/master'

lululwtv committed:
Merge remote-tracking branch 'origin/master'

lululwtv committed:
Merge remote-tracking branch 'origin/master'
